File: Interviews/LeetCode/Daily-Problems/2022/October/Day-27/835.cpp
class Solution {
public:
    int largestOverlap(vector<vector<int>>& img1, vector<vector<int>>& img2) {
        int n = img1.size();
        vector<int> pa, pb;
        for (int i = 0; i < n * n; i++) {
            if (img1[i / n][i % n] == 1) {
                pa.push_back((i / n) * 100 + (i % n));
            }
            if (img2[i / n][i % n] == 1) {
                pb.push_back((i / n) * 100 + (i % n));
            }
        }

        unordered_map<int, int> ump;
        for (int i : pa) {
            for (int j : pb) {
                ump[i - j] += 1;
            }
        }

        int ans = 0;
        for (auto i : ump) {
            ans = max(ans, i.second);
        }

        return ans;
    }
};

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-27/835.py
class Solution:
    def largestOverlap (self, img1: List[List[int]], img2: List[List[int]]) -> int:
        n, pa, pb, ump, ans = len(img1), [], [], {}, 0
        for i in range(n * n):
            if img1[i // n][i % n] == 1: pa.append((i // n) * 100 + (i % n))
            if img2[i // n][i % n] == 1: pb.append((i // n) * 100 + (i % n))
        
        for i in pa: 
            for j in pb: 
                ump[i - j] = ump.get(i - j, 0) + 1
        for value in ump.values(): ans = max(ans, value)
        
        return ans